waku/node/waku_node.nim

```diff
@@ -415,25 +443,96 @@ proc filterSubscribe*(node: WakuNode, pubsubTopic: Option[PubsubTopic], contentT
+      info "subscribed to topic", pubsubTopic=pubsub, contentTopics=topics
+
+proc filterSubscribe*(node: WakuNode,
+                        pubsubTopic: Option[PubsubTopic],
+                        contentTopics: ContentTopic|seq[ContentTopic],
+                        peer: RemotePeerInfo|string):
+
+                Future[FilterSubscribeResult]
+
+                {.async, gcsafe, raises: [Defect, ValueError].} =
+
+  ## Registers for messages that match a specific filter. Triggers the handler whenever a message is received.
+  if node.wakuFilterClient.isNil():
+    error "cannot register filter subscription to topic", error="waku filter client is not set up"
+    return err(FilterSubscribeError.serviceUnavailable())
+
+  let remotePeerRes = parsePeerInfo(peer)
+  if remotePeerRes.isErr():
+    error "Couldn't parse the peer info properly", error = remotePeerRes.error
+    return err(FilterSubscribeError.serviceUnavailable("No peers available"))
+
+  let remotePeer = remotePeerRes.value
+
+  if pubsubTopic.isSome():
+    info "registering filter subscription to content", pubsubTopic=pubsubTopic.get(), contentTopics=contentTopics, peer=remotePeer.peerId
+
+    let subRes = await node.wakuFilterClient.subscribe(remotePeer, pubsubTopic.get(), contentTopics)
+    if subRes.isOk():
+      info "v2 subscribed to topic", pubsubTopic=pubsubTopic, contentTopics=contentTopics
+    else:
+      error "failed filter v2 subscription", error=subRes.error
+      waku_node_errors.inc(labelValues = ["subscribe_filter_failure"])
+
+    return subRes
+  else:
+    let topicMapRes = parseSharding(pubsubTopic, contentTopics)
+
+    let topicMap =
+      if topicMapRes.isErr():
+        error "can't get shard", error=topicMapRes.error
+        return err(FilterSubscribeError.badResponse("can't get shard"))
+      else: topicMapRes.get()
+
+    var futures = collect(newSeq):
+      for pubsub, topics in topicMap.pairs:
+        info "registering filter subscription to content", pubsubTopic=pubsub, contentTopics=topics, peer=remotePeer.peerId
+        let content = topics.mapIt($it)
+        node.wakuFilterClient.subscribe(remotePeer, $pubsub, content)
+
+    let finished = await allFinished(futures)
+
+    var subRes: FilterSubscribeResult = FilterSubscribeResult.ok()
     for fut in finished:
       let res = fut.read()
 
       if res.isErr():
         error "failed filter subscription", error=res.error
         waku_node_errors.inc(labelValues = ["subscribe_filter_failure"])
+        subRes = FilterSubscribeResult.err(res.error)
 
     for pubsub, topics in topicMap.pairs:
       info "subscribed to topic", pubsubTopic=pubsub, contentTopics=topics
 
-proc filterUnsubscribe*(node: WakuNode, pubsubTopic: Option[PubsubTopic], contentTopics: ContentTopic|seq[ContentTopic],
-                  peer: RemotePeerInfo|string) {.async, gcsafe, raises: [Defect, ValueError].} =
-  ## Unsubscribe from a content filter.
+    # return the last error or ok
+    return subRes
```

Comment:
This is in combination with the two previous suggestions. _GH_ didn't allow me to make it in only one.
```suggestion
    return ok()
```